libbeat/_meta/fields.common.yml

```diff
@@ -15,6 +15,11 @@
       description: >
         The hostname as returned by the operating system on which the Beat is
         running.
+      metricbeat.format: url
```

Comment:
This is the only solution that I found to set a different format for `beat.hostname` for Metricbeat than for the other beats. Other option would be to remove it from common libbeat fields, but this is a bigger change.